backend/tasks/training_packs/binary_reader.py

```diff
@@ -0,0 +1,385 @@
+"""
+Sourced from NEO Python Core
```

Comment:
Can you add a URL to where it is?
(And make sure licensing is correct)

Reply:
Also are we sure this does not exist as a pypi library?

Reply:
It doesn't, it's part of their codebase. I'll add the license.